### ISSUE TITLE
metadata.max.age.ms :  description correction

### ISSUE THREAD

**Issue by nagaprabhu** [edenhill/librdkafka#1833]
Description
===========
Possible documentation correction:
    The description for  "metadata.max.age.ms" says "Metadata cache max age. Defaults to metadata.refresh.interval.ms * 3".

I don't see a property called "metadata.refresh.interval.ms". Do you mean  topic.metadata.refresh.interval.ms?

1. Tried to set metadata.refresh.interval.ms. That didn't reflect in the config dump. May be not a valid property.

2. Tried to set topic.metadata.refresh.interval.ms. I am able to see the new property value in config dump, but it didn't have any impact on metadata.refresh.interval.ms property.

How to reproduce
================
https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:

 - [x] librdkafka version (release number or git tag): `<f26b9b244945b3260ca40814df22ccd6ee779e80>`

**Comment by edenhill** [comment]
You're right in that `metadata.refresh.interval.ms` doesn't exist and should be `topic.metadata.refresh.interval.ms` (I'll fix this).

The `metadata.max.age.ms` does not directly reflect the refresh interval, only the cache eviction time, so it isn't always obvious when it kicks in.

What problem are you seeing exactly?

Are you saying that the metadata age

**Comment by nagaprabhu** [comment]
Hi Edenhill,

Thx for the quick action.

librdkafka Version : 0.9.4.1 (yup, slightly outdated)
Mode : Producer
Broker version : 0.10.2.0

Everything was fine for quiet sometime as long as I use 1 topic in 1 producer object. Recently, I started reusing the same producer object to pump data for one more topic ( so 2 topic in 1 producer object). since then, sporadically, I was getting "Message timed out"  in my callback ( even though I set message.timeout.ms to 0). On the consumreside messages were delayed by ~2 mins. This was happening for both topics. In this case, my metadata.max.age.ms was 1mins.

I just migrated to librdkafka (0.11.4) and configured metadata.max.age.ms to 5 mins. I dont see this issue anymore. 

During the analysis/migration process  to the latest librdkafka. I observed this documentation mismatch.

I don't have any active issue.
